### PR TITLE
Query Cache: Only cache initial query

### PR DESCRIPTION
**Pull request by zhongyuankai**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/executeQuery.cpp [hunk 1]
```diff
@@ -988,7 +988,11 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
         }
 
         QueryCachePtr query_cache = context->getQueryCache();
-        const bool can_use_query_cache = query_cache != nullptr && settings.use_query_cache && !internal && (ast->as<ASTSelectQuery>() || ast->as<ASTSelectWithUnionQuery>());
+        const bool can_use_query_cache = query_cache != nullptr
+            && settings.use_query_cache
+            && !internal
```
Comment (rschu1ze) [inline src/Interpreters/executeQuery.cpp:993]: I am confused. Does `internal` (l. 993) not correspond exactly to `client_info.query_kind == ClientInfo::QueryKind::INITIAL_QUERY`?

Comment (zhongyuankai) [reply src/Interpreters/executeQuery.cpp:993]: According to my understanding, `internal` distinguishes whether the current query is started by the current node, while `client_info.query_kind == ClientInfo::QueryKind::INITIAL_QUERY` distinguishes whether the current query is a subquery started by a distributed table. Actually I just want to cache the initial query.

Comment (zhongyuankai) [reply src/Interpreters/executeQuery.cpp:993]: According to my understanding, `internal` distinguishes whether the current query is started by the current node, while `client_info.query_kind == ClientInfo::QueryKind::INITIAL_QUERY` distinguishes whether the current query is a subquery started by a distributed table. Actually I just want to cache the initial query.

Comment (zhongyuankai) [reply src/Interpreters/executeQuery.cpp:993]: In some scenarios, you may want to cache the subqueries of the distribution table. I'm not sure.

**Pull request merged**